exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1349 @@
+			orderSide, err = order.StringToOrderSide(responseOrders.Items[i].Side)
+			if err != nil {
+				return nil, err
+			}
+			var orderStatus order.Status
+			pair, err = currency.NewPairFromString(responseOrders.Items[i].Symbol)
+			if err != nil {
+				return nil, err
+			}
+			pair.Delimiter = currency.DashDelimiter
+			var oType order.Type
+			oType, err = order.StringToOrderType(responseOrders.Items[i].Type)
+			if err != nil {
+				log.Errorf(log.ExchangeSys, "%s %v", ku.Name, err)
+			}
+			orders[i] = order.Detail{
+				Price:           responseOrders.Items[i].Price,
+				Amount:          responseOrders.Items[i].Size,
+				ExecutedAmount:  responseOrders.Items[i].DealSize,
+				RemainingAmount: responseOrders.Items[i].Size - responseOrders.Items[i].DealSize,
+				Date:            responseOrders.Items[i].CreatedAt.Time(),
+				Exchange:        ku.Name,
+				OrderID:         responseOrders.Items[i].ID,
+				Side:            orderSide,
+				Status:          orderStatus,
+				Type:            oType,
+				Pair:            pair,
+			}
+			orders[i].InferCostsAndTimes()
+		}
+	}
+	order.FilterOrdersBySide(&orders, getOrdersRequest.Side)
+	order.FilterOrdersByType(&orders, getOrdersRequest.Type)
+	err = order.FilterOrdersByTimeRange(&orders, getOrdersRequest.StartTime, getOrdersRequest.EndTime)
+	if err != nil {
+		log.Errorf(log.ExchangeSys, "%s %v", ku.Name, err)
+	}
+	order.FilterOrdersByPairs(&orders, getOrdersRequest.Pairs)
+	return getOrdersRequest.Filter(ku.Name, orders), nil
+}
+
+// GetFeeByType returns an estimate of fee based on the type of transaction
+func (ku *Kucoin) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	if feeBuilder == nil {
+		return 0, fmt.Errorf("%T %w", feeBuilder, common.ErrNilPointer)
+	}
+	if !ku.AreCredentialsValid(ctx) && // Todo check connection status
+		feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		feeBuilder.FeeType = exchange.OfflineTradeFee
+	}
+	switch feeBuilder.FeeType {
+	case exchange.CryptocurrencyWithdrawalFee,
+		exchange.CryptocurrencyTradeFee:
+		fee, err := ku.GetBasicFee(ctx, "0")
+		if err != nil {
+			return 0, err
+		}
+		if feeBuilder.IsMaker {
+			return feeBuilder.Amount * fee.MakerFeeRate, nil
+		}
+		return feeBuilder.Amount * fee.TakerFeeRate, nil
+	case exchange.OfflineTradeFee:
+		return feeBuilder.Amount * 0.001, nil
+	case exchange.CryptocurrencyDepositFee:
+		return 0, nil
+	default:
+		if !feeBuilder.FiatCurrency.IsEmpty() {
+			fee, err := ku.GetBasicFee(ctx, "1")
+			if err != nil {
+				return 0, err
+			}
+			if feeBuilder.IsMaker {
+				return feeBuilder.Amount * fee.MakerFeeRate, nil
+			}
+			return feeBuilder.Amount * fee.TakerFeeRate, nil
+		}
+		return 0, fmt.Errorf("can't construct fee")
+	}
+}
+
+// ValidateCredentials validates current credentials used for wrapper
+func (ku *Kucoin) ValidateCredentials(ctx context.Context, assetType asset.Item) error {
+	_, err := ku.UpdateAccountInfo(ctx, assetType)
```

Comment:
Please be aware that Kucoin has seperate API credentials for SPOT and FUTURES and in our current system, cannot support both. 
The way this is currently behaving means that we cannot have _any_ authenticated support because we're checking for both asset types and one of them fails